### PR TITLE
Disable the certificate validation instead of adding the certificate to the system

### PR DESCRIPTION
**Pull request by System25**
An untrusted SSL certificate can be untrusted for several reasons:

- Is a self-signed certificate.
- The name of the host of the certificate does not match the URL that you are connecting to. For example the name of a self-signed certificate use to be "localhost", and when you try to connect to that host by using https://<ip>/ (for example: https://192.168.1.20/)  that URL ("192.168.1.20" in the example) doesn't match "localhost".
- Is a certificate signed by a untrusted root certificate.

And when the certificate is signed by an untrusted root certificate is that root certificate the one that must the added to the system. That means that the computer must be able to download that root certificate from internet, and when the computer doesn't have internet connection (it only has a LAN connection) that is impossible.

So, due to all this causes, we considered that a better solution is to disable the certificate validation when the certificate is not trusted.

### CODE REVIEW

File: gecosws_config_assistant/controller/AutoSetupController.py [hunk 1]
```diff
@@ -141,23 +138,12 @@ def _setup_validate_data(self, gecosAccessData):
...
-                            if not sslUtil.isServerCertificateTrusted(gecosAccessData.get_url()):
-                                # Trusted certificates produce connection errors
-                                # due to many things. For example a bad name.
-                                errormsg = sslUtil.getUntrustedCertificateCause( gecosAccessData.get_url() )
-                                self.logger.debug("Error connecting to HTTPS server: %s"%(errormsg))
-                                showerror_gtk(_("Can't connect to GECOS CC!") + "\n" +  _("SSL ERROR:") + ' ' + errormsg,
-                                     None)
-                                self.view.focusUrlField()            
-                                return False                    
+                            SSLUtil.disableSSLCertificatesVerication()
```
Comment (alfem) [inline gecosws_config_assistant/controller/AutoSetupController.py]: "Verication" mistyped

File: data/templates/client.rb [hunk 2]
```diff
@@ -5,6 +5,8 @@
 
 log_location        STDOUT
 
+ssl_verify_mode    ${ssl_certificate_verication}
```
Comment (alfem) [inline data/templates/client.rb:8]: "verication" mistyped

File: gecosws_config_assistant/controller/ConnectWithGecosCCController.py [hunk 3]
```diff
@@ -462,29 +446,16 @@ def connect(self):
...
-                                    errormsg = sslUtil.getUntrustedCertificateCause( gem_repo )
-                                    self.logger.debug("Error connecting to HTTPS server: %s"%(errormsg))
-                                    self.processView.setChefCertificateRetrievalStatus(_('ERROR'))
-                                    self.processView.enableAcceptButton()
-                                    showerror_gtk(_("Can't connect to GEMs repository!") + "\n" +  _("SSL ERROR:") + ' ' + errormsg,
-                                         None)
-                                    gecosCC.unregister_chef_node(self.view.get_gecos_access_data(), workstationData.get_node_name())
-                                    self._clean_connection_files_on_error()
-                                    return False                                
+                                SSLUtil.disableSSLCertificatesVerication()
```
Comment (alfem) [inline gecosws_config_assistant/controller/ConnectWithGecosCCController.py]: "Verication" mistyped

File: gecosws_config_assistant/controller/ConnectWithGecosCCController.py [hunk 4]
```diff
@@ -589,29 +557,16 @@ def connect(self):
...
+                            self.logger.debug("User don't want to add the HTTPS server certificate to the Trusted CA list!")
                             self.processView.setLinkToChefStatus(_('ERROR'))
                             self.processView.enableAcceptButton()
                             gecosCC.unregister_chef_node(self.view.get_gecos_access_data(), workstationData.get_node_name())
                             self._clean_connection_files_on_error()
                             return False                 
                             
                         else:
-                            sslUtil.addCertificateToTrustedCAs(certificate, True)
+                            SSLUtil.disableSSLCertificatesVerication()
```
Comment (alfem) [inline gecosws_config_assistant/controller/ConnectWithGecosCCController.py]: "Verication" mistyped

File: gecosws_config_assistant/controller/ConnectWithGecosCCController.py [hunk 5]
```diff
@@ -630,8 +585,14 @@ def connect(self):
         template.owner = 'root'
         template.group = 'root'
         template.mode = 00644
+        
+        ssl_certificate_verication = ':verify_peer'
```
Comment (alfem) [inline gecosws_config_assistant/controller/ConnectWithGecosCCController.py]: "verication" mistyped

File: gecosws_config_assistant/tests/util/SSLUtilTest.py [hunk 6]
```diff
@@ -43,9 +43,20 @@ def runTest(self):
         sslUtil.removeCertificateFromTrustedCAs(certificate)
 
         # Start the test
+        SSLUtil.disableSSLCertificatesVerication()
```
Comment (alfem) [inline gecosws_config_assistant/tests/util/SSLUtilTest.py]: "Verication" mistyped

File: gecosws_config_assistant/util/GecosCC.py [hunk 7]
```diff
@@ -86,7 +87,7 @@ def validate_credentials(self, data):
             user = data.get_login()
             password = data.get_password()
             r = requests.get(url, auth=(user,password), headers=headers, 
-                             verify=True, timeout=self.timeout)
+                             verify=SSLUtil.isSSLCertificatesVericationEnabled(), timeout=self.timeout)
```
Comment (alfem) [inline gecosws_config_assistant/util/GecosCC.py]: "Verication" mistyped

File: gecosws_config_assistant/util/SSLUtil.py [hunk 8]
```diff
@@ -305,9 +307,30 @@ def __init__(self):
         self.logger = logging.getLogger('SSLUtil')
         self.timeout = 120
 
+    @staticmethod
+    def disableSSLCertificatesVerication():
```
Comment (alfem) [inline gecosws_config_assistant/util/SSLUtil.py]: "Verication" mistyped

**Review by alfem — COMMENTED** [review]
Just change all mistyped "verication" to "verification" and the PR will be accepted.

**Comment by System25** [comment]
Two commits added.

**Review by alfem — APPROVED** [review]

**Pull request merged**